### ISSUE TITLE
Field annotations in a `dataclass_transform` decorated class are ignored.

### ISSUE THREAD

**Issue by yaxum62** [microsoft/pyright#6537]
Pyright reports a false error in following code with strict settings:

```python
from typing import Any, dataclass_transform

@dataclass_transform()
class Base:
    base: int

    def __init__(self, **unused:Any):
        ...

class Derived(Base):
    derived: str

Derived(base=42, derived="")
#       ^^^^ No parameter named "base"
```

Removing `@dataclass_transform()` line in example code (improperly) fix the issue.

Pylance extension version v2023.11.10

**Comment by erictraut** [comment]
Pyright is working correctly here. The base class that is decorated with `dataclass_transform` is assumed to implement dataclass-like functionality for its child classes, but it is not itself treated like a dataclass. Fields defined in this base class therefore do not appear in the synthesized `__init__` method for its child classes.

To make this work, you'd need to add another level to the class hierarchy.

```python
from typing import Any, dataclass_transform

@dataclass_transform()
class DataclassBase:
    pass

class Base(DataclassBase):
    base: int

    def __init__(self, **unused:Any):
        ...

class Derived(Base):
    derived: str

Derived(base=42, derived="")
```
